apis/cli/v1alpha1/cliplugin_types.go

```diff
@@ -0,0 +1,73 @@
+// Copyright 2021 VMware, Inc. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// ArtifactList contains an Artifact object for every supported platform of a version.
+type ArtifactList []Artifact
```

Comment:
Curious why `[]Artifact` needs to be a separate type?

Reply:
This is needed because it is getting used with map as shown below
```go
Artifacts map[string]ArtifactList `json:"artifacts"`
```

Not adding a separate type throws error while generating manifest `map values must be a named type, not *ast.ArrayType`